[HACDOCS-1016] Added ArgoCD plugin content

gtrivedi88 opened the pull request:
Purpose: To resolve [HACDOCS-1016](https://issues.redhat.com/browse/HACDOCS-1016)

Version for cherrypicking: NA

Content for preview:  [Output]()

SME review: @karthikjeeyar

karthikjeeyar commented on artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:
```diff
@@ -1,322 +1,268 @@
+    plugins:
+      - package: ./dynamic-plugins/dist/roadiehq-backstage-plugin-argo-cd-backend-dynamic
+        disabled: false
+        pluginConfig:
+          argocd:
+            username: "${ARGOCD_USERNAME}"
+            password: "${ARGOCD_PASSWORD}"
+            appLocatorMethods:
+              - type: 'config'
+                instances:
+                  - name: argoInstance1
+                    url: "${ARGOCD_INSTANCE1_URL}"
+                    token: "${ARGOCD_AUTH_TOKEN}"
```
Safe to remove this, it is not required.
```suggestion
```

karthikjeeyar commented on artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:
```diff
@@ -1,322 +1,268 @@
+                    token: "${ARGOCD_AUTH_TOKEN}"
+      - package: ./dynamic-plugins/dist/janus-idp-backstage-plugin-argocd
+        disabled: false
+        pluginConfig:
+          dynamicPlugins:
+            frontend:
+              janus-idp.backstage-plugin-argocd:
+                mountPoints:
+                  - mountPoint: entity.page.overview/cards
+                    importName: ArgocdDeploymentSummary
+                    config:
+                      layout:
+                        gridColumnEnd:
+                          lg: "span 8"
+                          xs: "span 12"
+                      if:
+                        allOf:
+                          - isArgocdAvailable
+                  - mountPoint: entity.page.cd/cards
+                    importName: ArgocdDeploymentLifecycle
+                    config:
+                      layout:
+                        gridColumn: '1 / -1'
+                      if:
+                        allOf:
+                          - isArgocdConfigured
```
we can remove this.
```suggestion
```

karthikjeeyar commented on artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:
```diff
@@ -1,322 +1,268 @@
 ----
 
-Also, you can use Helm to install the ArgoCD plugin and be insecure by default:
+=== Loading the Plugin in Backstage Showcase Application
```
Can we add this link to the backstage-showcase repo somewhere here https://github.com/janus-idp/backstage-showcase ?

karthikjeeyar replied:
I was thinking more about this section, this section is about running the backstage-showcase repo locally by some developers, I dont think we need this section here in RHDH at all, we can remove this section.

karthikjeeyar commented on artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:
```diff
@@ -1,322 +1,268 @@
+= Installation and Configuration of ArgoCD
 
-The ArgoCD plugin represents the current status of an application in your Backstage Catalog.
+You can use the ArgoCD plugin to visualize Continuous Delivery (CD) pipeline runs in OpenShift GitOps. This plugin provides a visual overview of the application’s status, deployment details, author of the pipeline run, commit message, and the container image promoted to production.
```
```suggestion
You can use the ArgoCD plugin to visualize the Continuous Delivery (CD) workflows in OpenShift GitOps. This plugin provides a visual overview of the application’s status, deployment details, commit message, author of the commit, container image promoted to environment and deployment history.
```

karthikjeeyar reviewed (commented):
Can you please add some screenshots of the plugin, so the users know what will be available to them post configuration ?

karthikjeeyar commented on artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:
```diff
@@ -1,322 +1,268 @@
 ----
 
-Also, you can use Helm to install the ArgoCD plugin and be insecure by default:
+=== Loading the Plugin in Backstage Showcase Application
```
```suggestion
```

karthikjeeyar commented on artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:
```diff
@@ -1,322 +1,191 @@
 
+[NOTE]
+====
+For additional deployment details, select an appropriate card or select the link icon (image:rhdh-plugins-reference/link.svg[width=12px, height=12px, align=center]) to open the deployment details in *argo*.
```
Can we break it into two points, bcoz they are different operations.
```suggestion
1. For additional deployment details, click on a card to open the sidebar.
2. Click on the link icon (image:rhdh-plugins-reference/link.svg[width=12px, height=12px, align=center]) to open the deployment details in *Argo CD*.
```

karthikjeeyar commented on artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:
```diff
@@ -1,322 +1,191 @@
+For additional deployment details, select an appropriate card or select the link icon (image:rhdh-plugins-reference/link.svg[width=12px, height=12px, align=center]) to open the deployment details in *argo*.
+==== 
 
+. (Optional) Alternatively, select *Overview* tab, and then from Deployment summary section, select an appropriate application to open the deployment details in *argo*.
```
```suggestion
 *Overview* tab contains the Deployment summary, select an appropriate application to open it in *Argo CD*.
```

Can we also add the screenshot of the Overview tab here ?


![image](https://github.com/user-attachments/assets/d3563423-dc01-4fe4-8297-1664a1d69357)

karthikjeeyar commented on artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:
```diff
@@ -1,322 +1,191 @@
-= ArgoCD plugin for {product}
+= Installation and Configuration of ArgoCD
```
Can we use the name to Argo CD with a space throughout this doc?  Even in backstage-plugins repo, there is a PR to update the naming with a space. 

```suggestion
= Installation and Configuration of Argo CD plugin
```

karthikjeeyar reviewed (approved):
Thanks @gtrivedi88 for all the updates.

/lgtm

karthikjeeyar commented on artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:
```diff
@@ -1,322 +1,114 @@
-=== Permissions in ArgoCD plugin
-
-Setting permissions for ArgoCD user account not only reduces the scope but also reduces the functionality of the backend. If you scope the permissions for read-only, then actions including creating, deleting, and resyncing the application will not be available. 
+* Select an appropriate card. The system displays the deployment details (for example, commit history, author name, and deployment history) in a sidebar.
```
```suggestion
* Select an appropriate card. The system displays the deployment details (for example, commit message, author name, and deployment history) in a sidebar.
```

karthikjeeyar commented on artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:
```diff
@@ -1,322 +1,114 @@
-* Your Backstage application is installed and deployed.
-
-.Procedure
+* Add Argo CD instance information to your `app.config.yaml` file:
```
```suggestion
* Add Argo CD instance information to your `app.config.yaml` configmap:
```

subhashkhileri commented on artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:
```diff
@@ -1,322 +1,114 @@
-* Your Backstage application is installed and deployed.
-
-.Procedure
+* AAdd Argo CD instance information to your `app.config.yaml` configmap:
```
There is a typo `AAdd`

subhashkhileri commented on artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:
```diff
@@ -1,322 +1,114 @@
---
+ annotations:
+   ...
+    # The Argo CD instance name used in `app.config.yaml`.
```
Update config file name `app.config.yaml` to `app-config.yaml`

subhashkhileri commented on artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:
```diff
@@ -1,322 +1,114 @@
 [NOTE]
 ====
-You can either use annotations or labels for a component.
+If you do not set this annotation, the Argo CD plugin defaults to the first Argo CD instance configured in `app.config.yaml`.
```
Update config file name app.config.yaml to app-config.yaml

Gerry-Forde commented:
/lgtm

linfraze commented on artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:
```diff
@@ -1,322 +1,114 @@
---
 
-. Send a GET HTTP request to ArgoCD’s `/session` endpoint using the username and password to acquire your authentication key.
+.Procedures
```
Procedure

linfraze commented on artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:
```diff
@@ -1,322 +1,114 @@
-          # dedicated username/password for this instance
-          username: ${ARGOCD_USERNAME_INSTANCE_2} # optional
-          password: ${ARGOCD_PASSWORD_INSTANCE_2} # optional
+. On the {product-very-short} platform, navigate to the *Catalog* and select an appropriate component.
```
Click the **Catalog** tab and select the component that you want to use.

gtrivedi88 replied:
> Click the Catalog tab and select the component that you want to use.
Updated but raised a question below. Click vs Select. Kept select as for now

linfraze commented on artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:
```diff
@@ -1,322 +1,114 @@
-A loop is created between deleting ArgoCD project and application to check the deletion of application occurring before the deletion of the project. If waitCycles is set to 25, then the loop can last for 75 seconds before erroring out.
-
-With the previous configuration, the ArgoCD plugin fetches the ArgoCD instances an application is deployed to and use the ArgoCD backend plugin (`backstage-plugin-argo-cd-backend`) to reach out to each ArgoCD instance based on the following mapping:
+. Select *CD* tab to view insights into deployments managed by Argo CD. 
```
Click the **CD** tab

gtrivedi88 replied:
@linfraze Thank you for your suggestion. I decided to use "Select" based on this:
![image](https://github.com/user-attachments/assets/920f94c6-8748-4d0e-a304-9ee07b7dec77)

Let me know if you disagree or have some other references.
Thanks again.

gtrivedi88 replied:
Based on this discussion - https://redhat-internal.slack.com/archives/C04P5S34GP2/p1722440235254279
...going with Select.

linfraze replied:
@gtrivedi88 that's fine. I appreciate you starting the discussion in the CCS channel. We all learn a lot that way.

linfraze commented on artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:
```diff
@@ -1,322 +1,114 @@
+image::rhdh-plugins-reference/sidebar.png[Sidebar]
 
-image::argo-cd-plugin-overview-card.png[ArgoCD overview card]
-
-The ArgoCD overview card displays application information including *NAME*, *INSTANCE*, *SYNC STATUS*, *HEALTH STATUS*, and *LAST SYNCED*.
---
-
-. To view the detailed application information, select an application from the ArgoCD overview card.
-+
---
-A pop-up containing detailed information about the application appears, along with a link to ArgoCD dashboard.
++ 
+* Select the link icon (image:rhdh-plugins-reference/link.svg[width=13px, height=13px, align=center]) to open the deployment details in Argo CD.
```
Remove For deployment details

Format this as step 3: Optional: Click the [object] to view its deployment details, for example, commit message, author, name, and deployment history.
Format this is as a sub-step to step 3: Optional: Click the link icon to open the deployment details in Argo CD.

Use _click_ and _select_ in accordance with IBM stlye guideance throughout the doc

gtrivedi88 replied:
> Remove For deployment details
Updated

> Format this as step 3: Optional: Click the [object] to view its deployment details, for example, commit message, author, name, and deployment history.
> Format this is as a sub-step to step 3: Optional: Click the link icon to open the deployment details in Argo CD.

Updated

> Use click and select in accordance with IBM stlye guideance throughout the doc
Raised question above

linfraze commented on artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:
```diff
@@ -1,322 +1,114 @@
 
-image::argo-cd-plugin-overview-card-details.png[ArgoCD overview card]
---
+. Select the *Overview* tab and navigate to the Deployment summary section to review the summary of your application's deployment across namespaces. Additionally, select an appropriate Argo CD app to open the deployment details in Argo CD, or select a commit from the Revision column to review the changes in GitLab or GitHub.
```
Click the **Overview** tab

If Deployment summary is a heading on the UI, then bold it. Same for Revision.

gtrivedi88 replied:
Thank you!
Not sure if I understand this. Deployment summary is a section name that the system displays on the Overview tab. Do we need to bold section names? ..same for column names, like Revision.

Please suggest.

gtrivedi88 replied:
Based on this discussion - https://redhat-internal.slack.com/archives/C04P5S34GP2/p1722440235254279
...going with Select.

If you feel this is not right, we can discuss more :)

linfraze replied:
@gtrivedi88 I was basing the recommendation to bold the UI elements on [this guidance](https://redhat-documentation.github.io/supplementary-style-guide/#user-interface-elements), but use your judgement

linfraze commented on artifacts/rhdh-plugins-reference/argocd/argocd-plugin-readme.adoc:
```diff
@@ -1,322 +1,114 @@
+[role="_additional-resources"]
+.Additional resources
 
+* For more information on dynamic plugins, refer link:https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.2/html/configuring_plugins_in_red_hat_developer_hub/rhdh-installing-dynamic-plugins[Dynamic plugin installation].
```
Change _refer_ to _see_ to align with the rest of the RHDH docs. It also looks like "to" is missing after "refer" for _refer to_

gtrivedi88 commented:
@linfraze Thank you for the comments. Raised a new PR addressing your comments: https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/pull/413
Also, requested some suggestions on your comments
